server/map_rpc_server.go

```diff
@@ -309,9 +308,21 @@ func (t *TrillianMapServer) getLeavesByRevision(ctx context.Context, mapID int64
 func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapLeavesRequest) (*trillian.SetMapLeavesResponse, error) {
 	ctx, spanEnd := spanFor(ctx, "SetLeaves")
 	defer spanEnd()
-
+	if req.Revision <= 0 {
+		return nil, status.Error(codes.FailedPrecondition, "revision must be > 0")
+	}
 	t.setLeafCounter.Add(float64(len(req.Leaves)), strconv.FormatInt(req.MapId, 10))
```

Comment:
Can we decide a general approach for when to use spans and increment counters in relations to precondition checks? It seems odd that in this PR we perform this check in between these operations.

My view is that context-free input errors (e.g. missing/invalid parameters) should be checked before counters and spans. Validation which requires context (e.g. tree with given ID can't be found) can happen after.

Reply:
We could decide this in a follow-up PR. Right now there are methods that either start a span right in the beginning (regardless of the checks), or not start it at all (as in `GetLeavesByRevisionNoProof`). The PR keeps this consistent.

Reply:
In the frame of this PR, I can put the `Revision <= 0` check below the counter increase if you prefer?

Reply:
I think incrementing the counter consistently at the start of the method has some merits. Let's do that for now.